src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -84,10 +93,21 @@ export function ThemeProvider({
     }
   }, [backgroundColor, color, isParentThemeProvider]);
 
-  const style = {...customProperties, ...(!isParentThemeProvider && {color})};
+  let style;
+
+  if (isParentThemeProvider) {
+    style = customProperties;
+  } else if (
+    !isParentThemeProvider &&
+    parentContext!.cssCustomProperties !== toString(customProperties)
+  ) {
+    style = {...customProperties, ...{color}};
+  } else {
+    style = {color};
+  }
 
   return (
-    <ThemeContext.Provider value={{...theme, textColor: color}}>
+    <ThemeContext.Provider value={{...theme}}>
```

Comment:
Now merging textColor in has been solved up there we don't need to spread and thus create a whole new object anymore

```suggestion
    <ThemeContext.Provider value={theme}>
```